demo-server/src/test/java/com/illia/file_holder/FileHolderTest.java

```diff
@@ -1,29 +1,23 @@
 package com.illia.file_holder;
 
 import com.illia.server.config.ServerConfig;
+import com.illia.server.file_holder.FileHandler;
 import com.illia.server.file_holder.FileHolder;
 import com.illia.server.file_holder.FileHolderImpl;
-import org.apache.commons.io.FileUtils;
-import org.junit.jupiter.api.AfterEach;
-import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.beans.factory.annotation.Value;
 import org.springframework.boot.test.context.SpringBootTest;
 import org.springframework.boot.test.mock.mockito.MockBean;
 import org.springframework.core.io.ByteArrayResource;
-import org.springframework.test.context.TestPropertySource;
+import org.springframework.test.annotation.DirtiesContext;
 
 import java.io.*;
-import java.net.URISyntaxException;
-import java.nio.charset.StandardCharsets;
-import java.nio.file.Files;
 import java.nio.file.Path;
 
 import static org.junit.jupiter.api.Assertions.*;
-import static org.mockito.Mockito.when;
+import static org.mockito.Mockito.*;
 
-@TestPropertySource("classpath:application-test.properties")
+@DirtiesContext(classMode = DirtiesContext.ClassMode.AFTER_EACH_TEST_METHOD)
```

Comment:
I've never seen it; what is this for?

Reply:
It cleans ApplicationContext (creates and injects bean) after each test method run. In my case, it's used for cleaning FileHolder, there are several test methods that 'save' files and if running all tests at once there will appear fails due to not empty fileHolder. If running every test method manually, no such error occurre.